src/config.rs

```diff
@@ -207,7 +207,21 @@ impl KdfSettings {
         }
     }
 
-    pub(crate) fn get_kdf(&self, seed: &[u8]) -> Box<dyn kdf::Kdf> {
+    /// For writing out a database, generate a new KDF seed from the settings and return the KDF
+    /// and the generated seed
+    pub(crate) fn get_kdf_and_seed(
+        &self,
+    ) -> Result<(Box<dyn kdf::Kdf>, Vec<u8>), getrandom::Error> {
+        let mut kdf_seed = vec![0; self.seed_size()];
```

Comment:
@sseemayer I think we can make the `seed_size` function private now.

Reply:
We could. Before going 1.0, I would want to look at the complete public interface and check what can be hidden.